tests/console/systemd_testsuite.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's systemd tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run testsuite included in systemd sources
+# Maintainer: Thomas Blume <tblume@suse.com>
+
+use base "consoletest";
+use warnings;
+use strict;
+use testapi;
+use utils;
+
+sub run() {
+    # install systemd testsuite
+    select_console 'root-console';
+    if (check_var('VERSION', '12-SP2')) {
+        zypper_call('ar http://download.suse.de/ibs/QA:/SLE12SP2/update/ systemd-testrepo');
+    }
+
+    elsif (check_var('VERSION', '12-SP3')) {
+        zypper_call('ar http://download.suse.de/ibs/QA:/SLE12SP3/update/ systemd-testrepo');
+    }
+    else {
+        my $version = get_var('VERSION');
+        my $distri  = get_var('DISTRI');
+        die "systemd testsuite tests not supported for $distri version $version";
+    }
+
+    zypper_call('--gpg-auto-import-keys ref');
+    zypper_call('in systemd-testsuite');
+
+    # run the testsuite test scripts
+    my $scriptrun = assert_script_run('cd /var/opt/systemd-tests; ./run-tests.sh --all 2>&1 | tee /tmp/testsuite.log', 300);
```

Comment:
That won't work. `assert_script_run` does not return anything but fails the test if the exit code is non-zero. You can implement a post_fail_hook though where you can put all the handling in case of failures. Take a look into tests/console/http_srv.pm for an example.